drivers/display/display_ili9xxx.c

```diff
@@ -139,26 +106,29 @@ static int ili9xxx_write(const struct device *dev, const uint16_t x,
 	if (desc->pitch > desc->width) {
 		write_h = 1U;
 		nbr_of_writes = desc->height;
+		mipi_desc.height = 1;
+		mipi_desc.buf_size = desc->pitch * data->bytes_per_pixel;
 	} else {
 		write_h = desc->height;
+		mipi_desc.height = desc->height;
+		mipi_desc.buf_size = desc->buf_size;
 		nbr_of_writes = 1U;
 	}
 
-	r = ili9xxx_transmit(dev, ILI9XXX_RAMWR, write_data_start,
-			     desc->width * data->bytes_per_pixel * write_h);
+	mipi_desc.width = desc->width;
+	mipi_desc.height = desc->height;
```

Comment:
Since the buffer descriptor is passed to the underlying controller, I would prefer that `mipi_desc.pitch` is given a valid value (mipi_desc.pitch = desc->width here) so that host controllers (like ours) can take advantage of that feature.

Reply:
Fixed, thanks